app/js/sagas/conceptSaga.js

```diff
@@ -76,3 +78,41 @@ export function* fetchAndSetConcept(action) {
 export function* fetchConcept() {
   yield takeEvery(FETCH_CONCEPT, fetchAndSetConcept);
 }
+
+export function* fetchAndSetLabResultsToDisplayConceptSet(action) {
+  const { conceptUuid } = action;
+
+  try {
+    // hacky, only goes 5 levels deep
+    const CONCEPT_SET_REP = 'custom:(uuid,set,setMembers:(uuid,set,setMembers:(uuid,set,setMembers:(uuid,set,setMembers:(uuid,set,setMembers:(uuid,set,conceptClass:(name)),conceptClass:(name)),conceptClass:(name)),conceptClass:(name)),conceptClass:(name)),conceptClass:(name))';
+
+    const response = yield call(conceptRest.getConcept, conceptUuid, CONCEPT_SET_REP);
+
+    let concepts = response.setMembers;
+
+    // flatten the concept set
+    while (concepts.some(c => c.set)) {
+      concepts = concepts.flatMap(c => (c.set ? [...c.setMembers, {
+        uuid: c.uuid,
+        set: false,
+        conceptClass: c.conceptClass,
+      }] : c));
+    }
+
+    // filter out all that aren't lab tests or tests
+    concepts = concepts.filter(c => c.conceptClass.name === 'Test' || c.conceptClass.name === 'LabSet');
+
+    // create a set for easy lookup
+    const conceptSet = new Set();
+    concepts.forEach(c => conceptSet.add(c.uuid));
+
+    yield put(setLabResultsToDisplayConceptSet(conceptSet));
+  } catch (e) {
+    // TODO actually do something in failure case
```

Comment:
I confirmed that if the GP doesn't represent a valid concept, everything still works fine.